### PR TITLE
Revised topics about nullable types

### LINKED ISSUE #6630: Problem with the statement 'The ?? operator defines a default value that is returned when a nullable type is assigned to a non-nullable type.'
The statement 'The ?? operator defines a default value that is returned when a nullable type is assigned to a non-nullable type.' may be misleading or inaccurate.  The ?? operator doesn't care what type of assignment is being made.  I think a better description would be to indicate that ?? is the null coalescing operator which per https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/null-coalescing-operator, simply "...returns the left-hand operand if the operand is not null; otherwise it returns the right hand operand."

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 2552ed95-c11d-4657-0dad-9e0cccba9ca4
* Version Independent ID: 326dd492-3549-0312-3d46-a179d3fe76c6
* Content: [Using Nullable Types (C# Programming Guide)](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/nullable-types/using-nullable-types)
* Content Source: [docs/csharp/programming-guide/nullable-types/using-nullable-types.md](https://github.com/dotnet/docs/blob/master/docs/csharp/programming-guide/nullable-types/using-nullable-types.md)
* Product: **dotnet-csharp**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### PR DESCRIPTION
**Pull request by pkulikov**
Revised two topics about nullable types (probably, the best way to review is side by side):
https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/nullable-types/
https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/nullable-types/using-nullable-types

Moved snippets to the samples repo: supported by dotnet/samples#205

Fixes #6630

### CODE REVIEW

**Comment by BillWagner** [comment]
closing and reopening to force a new build

**Review by BillWagner — CHANGES_REQUESTED** [review]
I really like the work you've done here @pkulikov 
I had a number of small suggestions, and then it's ready.

Thanks again for all the work.

File: docs/csharp/programming-guide/nullable-types/using-nullable-types.md [hunk 1]
```diff
@@ -1,97 +1,88 @@
...
+You also can compare a nullable type variable with `null` instead of using the `HasValue` property, as the following example shows:  
   
-## The Members of Nullable Types  
- Each instance of a nullable type has two public read-only properties:  
+[!code-csharp-interactive[use comparison with null](../../../../samples/snippets/csharp/programming-guide/nullable-types/NullableTypesUsage.cs#3)]
   
--   `HasValue`  
+## Explicit conversions
+
+A nullable type can be cast to a regular type, either explicitly with a cast, or by using the `Value` property. For example:
```
[the remark this answers is not in the document]
Comment (pkulikov) [reply docs/csharp/programming-guide/nullable-types/using-nullable-types.md]: @BillWagner then the **Explicit conversions** section starts to overlap with the **Assignment to non-nullable type** section, which is further in the topic. I suggest to merge those two then and name it **Assignment to non-nullable type**.

Furthermore, the **Implicit conversions** section can be removed, as it almost duplicates the first section: **Declaration and assignment**.

That way, the whole topic would become less fragmented containing only the following sections:
 - Declaration and assignment of nullable type
 - Assignment to non-nullable type
 - Operators
 - The bool? type

What do you think?

**Comment by pkulikov** [comment]
@BillWagner thank you for the review. I've addressed your feedback for the *index.md* file. As for the *using-nullable-types.md* file, I've given it one more look and updated the division into sections. Please review it again.

The latest commit relies on the dotnet/samples#209

**Comment by pkulikov** [comment]
@BillWagner I've updated the "Using nullable types" topic with pattern matching case.

**Comment by BillWagner** [comment]
closing and reopening to start a new build, to include the new sample)

**Review by BillWagner — APPROVED** [review]

**Comment by BillWagner** [comment]
Thanks for making the updates @pkulikov I've reviewed the latest changes, and I'll :shipit: now.

Thanks again.

**Pull request merged**